src/nccl_ofi_sendrecv.c

```diff
@@ -467,6 +467,29 @@ static int post_recv_conn(nccl_net_ofi_sendrecv_listen_comm_t *l_comm,
 	return rc;
 }
 
+/*
+ * @brief	Returns the domain, dependent on the platform.
+ *
+ * @return	fid_domain for the device (P-series) or endpoint (Neuron).
+ * @return	0 on success
+ *
+ */
+
+static int get_domain(nccl_net_ofi_ep_t *base_ep, int rail, struct fid_domain **domain)
+{
+	nccl_net_ofi_sendrecv_ep_t *ep =
+		(nccl_net_ofi_sendrecv_ep_t *)base_ep;
+#if HAVE_NEURON
```

Comment:
You could more easily handle the Neuron distinction by just making `ep->domain` a backpointer to `device->domain` in the `HAVE_CUDA` case. Then `get_domain` would always just be `*domain = ep->domain`.

Reply:
If we do it that way, is it worth having a get_domain() method at all?